code/API_definitions/number-verification.yaml

```
@@ -2,43 +2,66 @@ openapi: 3.0.3
-
+    - **Network-Based Authentication**: Authentication mechanism based on the identification of the endpoint of a network connection.
+    Network operators know to which subscriber a network resource is assigned at a given moment, for example the mobile phone number associated to a specific mobile network connection.
+    - **SIM-Based Authentication**: Authentication mechanism based on the identification of the subscriber's SIM installed in the user's device.
     # API Functionality
 
-    This enables a Service Provider (SP) to verify the phone number of the mobile device being used to access their service where the mobile device is accessing the *service provider* over a mobile network (WiFi connections are out of this API scope). This can happen either by getting the comparison result or receiving the phone number of the device that is used, so they can verify it themselves.
+    This API enables an API Consumer to verify or retrieve the phone number of the mobile device being used to access their service
+
+    # The Authentication Request
+
+    If the API Consumer has a TS.43 temporary token created on the mobile device then this API works over all connections e.g. WiFi taking advantage of the SIM-Based authentication.
+    How the API Consumer gets a TS.43 temporary token is out-of-scope of the API definition.
+    When the API Consumer sends the Authentication Request, the temporary token is send in the login_hint parameter with the prefix "operatortoken:".
+
+    If the API Consumer does not have a TS.43 temporary token then the API Consumer must use OpenId Connect Authorization Code Flow as described in current release of [CAMARA APIs Access and User Consent Management](https://github.com/camaraproject/IdentityAndConsentManagement).
+    This mechanism of utilizing the mobile connection does not work on a fixed-net Internet connection nor over WiFi or VPN.
+    The API Consumer should not send a login_hint in the Authentication Request.
 
     # Resources and Operations overview
-    This API currently provides two endpoints which both require a **3-legged token** and authentication via **mobile network** (**excluding** for example by SMS/OTP or user/password as an authentication method):
-    - The first one checks if the user mobile phone number matches the phone number associated with the mobile device. It can receive either a hashed or a plain text phone number as input and it compares the received input with the authenticated user's phone number associated to the access token in order to respond **true/false**.
-    - The next one retrieves the phone number associated to the user's token and returns it so the verification can be made by the service provider.
+
+    This API currently provides two endpoints which both require a **3-legged token**:
```

I think it is important to not remove this part since it explicitly states what authentication methods cannot be used

```suggestion
    This API currently provides two endpoints which both require a **3-legged token** (**excluding** for example by SMS/OTP or user/password as an authentication method):
```

---

Maybe we can append to the sentence the following: "using one of the two methods indicated in _The Authentication Request_ section"

---

We already have text or that:
> Authentication methods such as SMS OTP or user/password are incompatible,

---

my suggestion was to refer to the two methods allowed now in the API, but the sentence about SMS/OTP or user/password is the opposite, mention what is not allowed. I'd leave the text as follows

This API currently provides two endpoints which both require a **3-legged token** obtained by using one of the two methods indicated in _The Authentication Request_ section (therefore **excluding** for example by SMS/OTP or user/password as an authentication method):

---

Diego's proposal to make it simpler to see/merge:

```suggestion
    This API currently provides two endpoints which both require a 3-legged token obtained by using one of the two methods indicated in The Authentication Request section (therefore excluding for example by SMS/OTP or user/password as an authentication method):
```